articles/quickstart/native/ios-objc/03-user-sessions.md

```diff
@@ -0,0 +1,282 @@
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+[keychain setString:accessToken forKey:@"access_token"];
+```
+
+As you can see, `SimpleKeychain` can be seen simply as a key-value storage.
+
+## On Startup: Check accessToken existence
+
+The main purpose of storing this token is to save the user from having to re-enter login credentials upon relaunch of the app. So, **once the app has launched**, we need to check for the existence of an `accessToken` to see if we can automatically log the user in and redirect the user straight into the app's main flow, skipping any login screen.
+
+To do so, first, you retrieve its value from the `accessToken` key stored in the keychain:
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+NSString* accessToken = [keychain stringForKey:@"access_token"];
+if (accessToken) {
+    // accessToken exists
+    // You still need to validate it!
+}
+```
+
+## Validate an accessToken
+
+Then, if such a token exists, you need to check whether it's still valid, has expired, or is no longer valid for some other reason, such as being revoked. To do so, you will use `Auth0` to fetch the user's profile based on the `accessToken` we've got:
+
+```objc
+[auth userInfoWithAccessToken:accessToken callback:^(NSError * _Nullable error, A0Profile * _Nullable profile) {
+        if (error) {
+            // accessToken has expired or no longer valid
+        } else {
+            // The accessToken is still valid and you have the user's profile
+            // This would be a good time to store the profile
+        }
+}];
+```
+
+## Dealing with a non-valid accessToken
+
+How to deal with a non-valid accessToken is up to you. You will normally choose between two scenarios:
+Either you ask users to re-enter theirs credentials, or you can use `.renew(withRefreshToken: refreshToken)` with a [refresh_token((/refresh-token)) to obtain a new valid accessToken again.
+
+If you aim for the former scenario, make sure you clear all the keychain stored values by doing:
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+[keychain clearAll];
+```
+
+However, in this tutorial you will focus on the latter scenario, where you want to log users in without asking for their credentials again.
+
+In this case, you're going to leverage the `refreshToken`. The refresh token is another token string contained within the `Credentials` object that comes upon a successful login, which doesn't expire, and whose main purpose is retrieving a new valid `accessToken`.
+
+::: note
+It's recommended that you read and understand the [refresh token documentation](/refresh-token) before proceeding. **You got to keep in mind, for example, that, even though the refresh token cannot expire, it can be revoked.**
```

Comment:
Slight re-wording: `It's recommended that you read and understand the [refresh token documentation](/refresh-token) before proceeding. You should keep in mind, for example, that even though the refresh token cannot expire, it can be revoked.`